src/viking.js

```diff
@@ -1,11 +1,86 @@
 // Soldier
-function Soldier() {}
+function Soldier (health, strength) {
+    this.health = health;
+    this.strength = strength;
+    this.attack = function () { 
+        return this.strength;
+    };
+    this.receiveDamage = function (damage) {
+        this.health -= damage;
+    };
+    }
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+
+    Soldier.call(this, health, strength);
+    this.name = name;
+    this.receiveDamage = function (damage) {
+      this.health -= damage;
+      if (this.health > 0) {
+        return `${this.name} has received ${damage} points of damage`;
+      } else {
+        return `${this.name} has died in act of combat`;
+      }
+    };
+    this.battleCry = function () {
+      return `Odin Owns You All!`
+    }
+  }
+  
+// I DON'T UNDERSTAND THE FOLLOWING STATEMENT. It was created by my partner, Henriette, on day 5:
+  Viking.prototype = Object.create(Soldier.prototype);
+  Viking.prototype.constructor = Viking;
 
 // Saxon
-function Saxon() {}
+function Saxon(health, strength) {
+
+    Soldier.call(this, health, strength);
+    this.receiveDamage = function (damage) {
+        this.health -= damage;
+        if (this.health > 0) {
+          return `A Saxon has received ${damage} points of damage`;
+        } else {
+          return `A Saxon has died in combat`;
+        }
+      }
+}
+
+// I DON'T UNDERSTAND THE FOLLOWING STATEMENT. It was created by my partner, Henriette, on day 5:
+Saxon.prototype = Object.create(Soldier.prototype);
+Saxon.prototype.constructor = Saxon;
 
 // War
-function War() {}
+function War() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+};
+
+War.prototype.addViking = function (viking) {
+    this.vikingArmy.push(viking);
+};
+
+War.prototype.addSaxon = function (saxon) {
+    this.saxonArmy.push(saxon);
+};
+
+/*
+War.prototype.vikingAttack() {
+    // ???
```

Comment:
You need to do call back the functions that should happen in case of viking attack, so probably Viking.attack() and Saxon.receiveDamage()